### ISSUE TITLE
Reintroduce the manual sorting of columns?

### ISSUE THREAD

**Issue by fredrmag** [LCBC-UiO/noas#41]
Now, there is no way of "manually" sort the columns. Before this could be specified in the metadata.json, but it was removed in this commit: 
https://github.com/LCBC-UiO/noas/commit/0b3e36beb566e639de7befe25e98f35331eb7a06#diff-30d42f64ebf4b427c1812ee65bf[…]ae483a5e19baa80da497ad74823fea48

**Comment by f-krull** [comment]
What's the use case? 
Currently, the columns are sorted alphabetically by their id. 
Yes, previously the idx allowed a custom order. The reason for removal of idx was to have the ordering consistent across all tables

**Comment by fredrmag** [comment]
E.g now, I added PET data, and they have a date and time column that I
would like to come before the statistics when stuff is displayed, and when
you retrieve it

ons. 12. mai 2021 kl. 15:23 skrev f-krull ***@***.***>:

**Comment by f-krull** [comment]
Is PET some timeseries data (type=repeated in noas). Then the 4th column gets sorted to the top. That is 
because in "repeated" tables the 4th column is always part of the tables primary key, and therefore the 4th column currently considered more important than the other columns and placed at the top. 
Maybe an easy fix is to combine date and time into a datetime and place it at the top? (Only work if the table type is "repeated")

Otherwise prefixing the columns IDs somehow might give an order without having to change the code. 

If both are not an option, then it needs some thoughts. Placing an index directly into the metadata directly didn't seem like a good idea. At least I was very happy when it got removed. The UI should have a choice how to display the columns. (Maybe rebranding `idx` to `display_priority` or something.. thinking out loud...)

**Comment by fredrmag** [comment]
Nah, crossectional/longitudinal data.

Prefixing with a number or letter a will make it sort to the top, but then
I feel the variable name would be ugly.

I am thinking about idx as a display priority variable now. So that could
be an option

ons. 12. mai 2021 kl. 17:58 skrev f-krull ***@***.***>:

**Comment by f-krull** [comment]
Taking the column order directly from the tsv might be an option too. Seems cleaner than setting an index for each column.

Anyway, I see the use case of such a feature. But I think we need to find a way needs to find a way that rules out misuse. 
When we removed title and idx, the NOAS data became cleaner because both fields were sometimes set to kind of arbitrary values (Murphy's law?)

**Comment by Athanasiamo** [comment]
I also mentioned on slack where we first started this conversation that using the column order as in the file as a nice solution. I kind of prefer this, as this creates clean data structures in the raw data, rather than relying on meta-data for changing how things are displayed.

**Comment by fredrmag** [comment]
For the display priority, I was thinking that you do not need to give a value to each column, but only the columns you want to have priority over others. 

But, yeah, using the column order as in the raw file sounds like the cleanest solution, as suggested by @Athanasiamo on slack before, and now by @f-krull here.

Can we think of ways people could misuse this? Or why you would misuse it?

**Comment by f-krull** [comment]
> Can we think of ways people could misuse this? Or why you would misuse it?

Changing the column order in the tsv alone is fine and encouraged. Also proving this info to the UI seem ok. 

But how would you change the UI? Order columns as in tsv by default?

**Comment by fredrmag** [comment]
Yeah, I would think if we go for the solution of sorting the data by the columns in the raw data, then that is default. 

Or, thinking out load, we could have a field in the top of the `_metadata.json` that can decide how you would want it to be sorted. E.g., `"order": "raw"`, or `"order": "alphabetical"`.

**Comment by Athanasiamo** [comment]
I'd rather have the data be stored cleverly and not needing any sorting of data afterwards. I think this is the cleaner long-term solution.

**Comment by f-krull** [comment]
Sounds good. Then we do the cleanup of all tables in NOAS (new issue?). Once that is done we change to columns order as in TSV?

**Comment by f-krull** [comment]
@fredrmag New idea: If the PET table contains lots of columns and the "date", "time" columns should be emphasized:

Create a PET info table and link it (as with mri).

**Comment by Athanasiamo** [comment]
We were hoping to avoid that with PET. Its not as complicated as MRI (we have little, and only by accident is it longitudinal). So I kind of want it all in a single table.

**Comment by f-krull** [comment]
I see. But let's not rush features and instead appreciate the fact that we have a UI in the first place ;)

**Comment by Athanasiamo** [comment]
no need to rush it I agree. I might suggest we table this feature until after the summer, if you are ok with that @fredrmag ? 
The UI is a splendid thing :)

**Comment by f-krull** [comment]
Reordering the tables might take a bit anyway

**Comment by fredrmag** [comment]
I am very ok with not rushing!:) It was just for convenience that I brought this up.

**Comment by Athanasiamo** [comment]
ok, we'll be dealing with this more directly in #42
